nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/actions/CSVExportAction.java

```diff
@@ -0,0 +1,309 @@
+            String commandId = commandIdFrom(record);
+            appendToFile(commandId, dataFrom(record), countFrom(record));
+            if (isFinished(commandId)) {
+                saveInTransientStore(context, commandId);
+            }
+        }
+
+        protected boolean isFinished(String commandId) {
+            KeyValueStore kvStore = Framework.getService(KeyValueService.class).getKeyValueStore(BULK_KV_STORE_NAME);
+            BulkStatus status = BulkCodecs.getBulkStatusCodec().decode(kvStore.get(commandId + STATUS));
+            return status.getCount() == count;
+        }
+
+        protected void appendToFile(String commandId, String content, long count) {
+            if (temp == null) {
+                try {
+                    temp = createTemp(commandId);
+                } catch (IOException e) {
+                    getLog().error(e, e);
+                    return;
+                }
+            }
+            try (FileOutputStream stream = new FileOutputStream(temp.toFile(), true)) {
+                stream.write(content.getBytes());
+                stream.flush();
+                this.count += count;
+            } catch (IOException e) {
+                getLog().error(e, e);
+            }
+        }
+
+        protected void saveInTransientStore(ComputationContext context, String commandId) {
+            storeBlob(new FileBlob(temp.toFile()), commandId);
+            try {
+                Files.delete(temp);
+            } catch (IOException e) {
+                getLog().error(e, e);
+            }
+            temp = null;
+            context.produceRecord("o1", BulkRecords.of(commandId, 1, getKey(commandId)));
+            context.askForCheckpoint();
+        }
+
+    }
+
+    public static class SortBlobLines extends AbstractTransientBlobComputation {
+
+        public static final String NAME = "sortBlobLines";
+
+        public SortBlobLines() {
+            super(NAME);
+        }
+
+        @Override
+        public void processRecord(ComputationContext context, String documentIdsStreamName, Record record) {
+            String commandId = commandIdFrom(record);
+            Blob blob = getBlob(dataFrom(record));
+            blob = sort(blob, commandId);
+            if (blob != null) {
+                storeBlob(blob, commandId);
+            }
+            context.produceRecord("o1", BulkRecords.of(commandId, 1, getKey(commandId)));
+            context.askForCheckpoint();
+        }
+
+        // use http://www.algosome.com/articles/how-to-sort-large-file.html
+        protected Blob sort(Blob blob, String commandId) {
+            File file = blob.getFile();
+            if (file == null) {
+                getLog().error(String.format("Blob for %s is not backed by a file, sort is not supported", commandId));
+                return null;
+            }
+            try (Stream<String> lines = Files.lines(file.toPath())) {
+                Path temp = createTemp(commandId);
```

Comment:
On windows, there's issue when overriding a file by move on FS. Not sure if it's relevant in BAF context.
Solution adopted in hot reload was to open current file and then write to it by overriding previous content.

Reply:
The same thing